crates/rules.bzl

```diff
@@ -0,0 +1,195 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+load("@rules_rust//rust:rust_common.bzl", "CrateInfo")
+
+CrateDeploymentInfo = provider(
+    fields = {
+        "crate": "Crate file to deploy",
+        "metadata": "File containing package metadata",
+    },
+)
+
+def _generate_version_file(ctx):
+    version_file = ctx.file.version_file
+    if not ctx.attr.version_file:
+        version_file = ctx.actions.declare_file(ctx.attr.name + "__do_not_reference.version")
+        version = ctx.var.get("version", "0.0.0")
+
+        ctx.actions.run_shell(
+            inputs = [],
+            outputs = [version_file],
+            command = "echo -n {} > {}".format(version, version_file.path),
+        )
+    return version_file
+
+def _assemble_crate_impl(ctx):
+    print(ctx.attr.target[CrateInfo])
```

Comment:
let's remove the print statement. this is only for debug right? if that's the case this is something that you should ideally find by reviewing your own PR before it gets to me.

Reply:
@lolski addressed in https://github.com/vaticle/bazel-distribution/pull/322/commits/1a5520b97a908e201634c169b07d8945b46fe254